### ISSUE TITLE
Reused nodes result in orphans not getting rescued

### ISSUE THREAD

**Issue by eprothro** [sorentwo/oban#862]
### Environment

- Oban 12.4.2
- Oban Pro 0.13.1
- PostgreSQL `postgis/postgis:11-2.5-alpine`
- Elixir 1.13.4 (compiled with Erlang/OTP 23)

### Current Behavior

With DynamicLifeline configured with the SmartEngine, the current default of the node name getting set to the hostname or`DYNO` (i.e. Heroku ) results in any orphaned jobs at process restart not being rescued.

Upon restarting the node(s) while jobs are executing, those jobs remain in the executing state but do not execute (i.e. they are orphaned). Once the new processes are up, they process new work, but the orphaned jobs remain "executing" but are not rescued (waited 5+ minutes).

### Expected Behavior

I expect that orphaned jobs will be rescued within 1 - 2 minutes without additional configuration, even if I am reusing nodes for execution (e.g. local development, EC2 instances, heroku deployment).

### Workaround

If I set the `node` name to ensure uniqueness, I see the expected behavior.

```
# config/config.runtime.exs
hostname = System.get_env("DYNO") || :inet.gethostname |> elem(1)

config :listing_engine, Oban,
  node: "#{hostname}/#{Timex.now()}"
```

Apologies for the noise if this is expected behavior. If it is, clarifying this in the documentation of `DynamicLifeline` may be helpful.

**Comment by sorentwo** [comment]
Thanks for closing. Your description doesn’t match how the smart engine works—namely, it isn’t based on node names.

Please share any other information you have so I can help you diagnose the problem (here or via email).

**Comment by eprothro** [comment]
Yes, I hesitated to even report it, but I definitely saw 5+ minutes of orphans not being rescued while new producers were up and processing other jobs. Posted in case there are any +1s out there. If I see it again I will dig in and report back.

The only other information I can offer right now is that the orphans were rescued at the same time another producer was added.

**Comment by sorentwo** [comment]
That sounds correct. The existing producer records aren't cleaned up until new producers start, and orphans won't be rescued as long as their producer looks alive.

**Comment by eprothro** [comment]
Sorry, to be clear, the sequence was:

1. jobs running on worker 1 and worker 2
2. worker 1 and 2 killed
3. dozens of jobs orphaned
4. worker 1 and worker 2 restart (new processes)
5. available jobs processed by worker 1 and 2, orphans remain orphaned
6. 5+ minutes go by
7. worker 3 spins up
8. orphans rescued

But like I said, it can't be just that sequence (there had to be some other factor I'm not remembering or didn't notice), as I can't reproduce.
